cookbook/agent_os/customize/middleware/extract_content_middleware.py

```diff
@@ -0,0 +1,184 @@
+"""Example for AgentOS to show how to extract metadata from headers."""
+
+import json
+
+from agno.agent import Agent
+from agno.db.sqlite.sqlite import SqliteDb
+from agno.os import AgentOS
+from fastapi import Request, Response
+from starlette.middleware.base import (
+    BaseHTTPMiddleware,
+)
+from starlette.middleware.base import (
+    _StreamingResponse as StreamingResponse,
+)
+
+
+# Setup Middleware
+class MetadataExtractionMiddleware(BaseHTTPMiddleware):
```

Comment:
Cool. Two separate examples could work better, if we want to show both extracting data from the headers and the "post-hook logic" (_send_notifications)